File: 2024/02/02.py
def check(levels):
    inc_or_dec = sorted(levels) in (levels, levels[::-1])
    rules = all(1 <= abs(a - b) <= 3 for a, b in zip(levels, levels[1:]))
    return rules and inc_or_dec

sum1 = sum2 = 0
for line in open(0):
    levels = list(map(int, line.split()))
    if check(levels):
        sum1 += 1
        sum2 += 1
    else:
        sum2 += any(check(levels[:i] + levels[i + 1 :]) for i in range(len(levels)))

print(sum1, sum2)

File: 2024/02/02.js
check = (levels) => {
    sorted = levels.slice().sort()
    inc_or_dec = [sorted.slice().reverse().join(), sorted.join()].includes(levels.join())
    rules = levels.every((a, i) => {
        b = levels[i - 1]
        return i == 0 || Math.abs(a - b) >= 1 && Math.abs(a - b) <= 3
    })
    return rules && inc_or_dec
}

sum1 = sum2 = 0
process.stdin.on('data', data => {
    data.toString().trim().split('\n').map(line => line.split(' ')).forEach(levels => {
        if (check(levels)) {
            sum1 += 1
            sum2 += 1
        } else {
            sum2 += levels.some((_, i) => check(levels.slice(0, i).concat(levels.slice(i + 1))))
        }
    })
    console.log(sum1, sum2)
})

